pinot-core/src/main/java/com/linkedin/pinot/core/segment/creator/impl/inv/OffHeapBitmapInvertedIndexCreator.java

```diff
@@ -238,26 +244,29 @@ public void seal() throws IOException {
       IOUtils.closeQuietly(fisOffsets);
       IOUtils.closeQuietly(fos);
       IOUtils.closeQuietly(fos);
-      // MMaputils handles the null checks for buffer
-      MmapUtils.unloadByteBuffer(origValueBuffer);
-      origValueBuffer = null; valueBuffer = null;
-      if (origLengths != null) {
-        MmapUtils.unloadByteBuffer(origLengths);
-        origLengths = null; lengths = null;
-      }
-      MmapUtils.unloadByteBuffer(origPostingListBuffer);
-      origPostingListBuffer = null; postingListBuffer = null;
-      MmapUtils.unloadByteBuffer(origPostingListCurrentOffsets);
-      origPostingListCurrentOffsets = null; postingListCurrentOffsets = null;
-      MmapUtils.unloadByteBuffer(origPostingListLengths);
-      origPostingListLengths = null; postingListLengths = null;
-      MmapUtils.unloadByteBuffer(origPostingListStartOffsets);
-      origPostingListStartOffsets = null; postingListStartOffsets = null;
+      releaseResources();
       FileUtils.deleteQuietly(new File(tempOffsetsFile));
       FileUtils.deleteQuietly(new File(tempBitmapsFile));
     }
   }
 
+  // Release all the memory that has been allocated in the constructor.
+  public void releaseResources() {
+    // MMaputils handles the null checks for buffer
+    MmapUtils.unloadByteBuffer(origValueBuffer);
+    origValueBuffer = null; valueBuffer = null;
+    MmapUtils.unloadByteBuffer(origLengths);
+    origLengths = null; lengths = null;
+    MmapUtils.unloadByteBuffer(origPostingListBuffer);
+    origPostingListBuffer = null; postingListBuffer = null;
+    MmapUtils.unloadByteBuffer(origPostingListCurrentOffsets);
+    origPostingListCurrentOffsets = null; postingListCurrentOffsets = null;
+    MmapUtils.unloadByteBuffer(origPostingListLengths);
+    origPostingListLengths = null; postingListLengths = null;
```

Comment:
What happens to the unloadByteBuffer call if one of the previous allocations failed and threw exceptions?

Reply:
all the fields are null to begin with, and unload handles the call with null buffer.